pallets/authors-manager/src/lib.rs

```diff
@@ -0,0 +1,634 @@
+
+    #[pallet::storage]
+    #[pallet::getter(fn get_ingress_counter)]
+    pub type TotalIngresses<T: Config> = StorageValue<_, IngressCounter, ValueQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        pub authors: Vec<(T::AccountId, ecdsa::Public)>,
+    }
+
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            Self { authors: Vec::<(T::AccountId, ecdsa::Public)>::new() }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+        fn build(&self) {
+            log::debug!(
+                "Authors Manager Genesis build entrypoint - total authors: {}",
+                self.authors.len()
+            );
+            for (author_account_id, eth_public_key) in &self.authors {
+                assert_ok!(<AuthorAccountIds<T>>::try_append(author_account_id));
+                <EthereumPublicKeys<T>>::insert(eth_public_key, author_account_id);
+            }
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(<T as Config>::WeightInfo::add_author())]
+        #[transactional]
+        pub fn add_author(
+            origin: OriginFor<T>,
+            author_account_id: T::AccountId,
+            author_eth_public_key: ecdsa::Public,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+            let author_account_ids = Self::author_account_ids().ok_or(Error::<T>::NoAuthors)?;
+            ensure!(!author_account_ids.is_empty(), Error::<T>::NoAuthors);
+
+            ensure!(
+                !author_account_ids.contains(&author_account_id),
+                Error::<T>::AuthorAlreadyExists
+            );
+            ensure!(
+                !<EthereumPublicKeys<T>>::contains_key(&author_eth_public_key),
+                Error::<T>::AuthorEthKeyAlreadyExists
+            );
+
+            ensure!(
+                AuthorAccountIds::<T>::get().unwrap_or_default().len() <
+                    (<MaximumAuthorsBound as sp_core::TypedGet>::get() as usize),
+                Error::<T>::MaximumAuthorsReached
+            );
+
+            Self::register_author(&author_account_id, &author_eth_public_key)?;
+
+            <AuthorAccountIds<T>>::try_append(author_account_id.clone())
+                .map_err(|_| Error::<T>::MaximumAuthorsReached)?;
+            <EthereumPublicKeys<T>>::insert(author_eth_public_key, author_account_id);
+
+            return Ok(())
+        }
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as Config>::WeightInfo::remove_author(MAX_AUTHOR_ACCOUNTS))]
+        #[transactional]
+        pub fn remove_author(
+            origin: OriginFor<T>,
+            author_account_id: T::AccountId,
+        ) -> DispatchResult {
+            let _ = ensure_root(origin)?;
+
+            Self::remove_deregistered_author(&author_account_id)?;
```

Comment:
Does the author need to be deregistered to remove or does this enacts the deregistration process?